### PR TITLE
chore(deps): Bump to syn 2, serde_with 3, darling 0.20, and serde_derive_internals 0.28

### PR DESCRIPTION
**Pull request by dsmith3197**

### CODE REVIEW

File: lib/vector-config-common/src/validation.rs [hunk 1]
```diff
@@ -336,17 +336,19 @@ fn maybe_float_or_int(meta: &Meta) -> darling::Result<Option<f64>> {
...
-            Lit::Str(s) => {
-                let s = s.value();
-                s.as_str()
-                    .parse()
-                    .map_err(|_| darling::Error::unknown_value(s.as_str()))
-            }
-            Lit::Int(i) => i.base10_parse::<f64>().map_err(Into::into),
-            Lit::Float(f) => f.base10_parse::<f64>().map_err(Into::into),
-            lit => Err(darling::Error::unexpected_lit_type(lit)),
+        Meta::List(_) => Err(darling::Error::unexpected_type("list")),
```
Comment (tobz) [inline lib/vector-config-common/src/validation.rs:339]: Oooh, nice catch here. 👍🏻

**Review by tobz — DISMISSED** [review]
Assuming it passes CI and none of the configuration schema output changes (i.e. it doesn't change the generated Cue docs) then this looks good to me. 👍🏻

**Comment by dsmith3197** [comment]
> Assuming it passes CI and none of the configuration schema output changes (i.e. it doesn't change the generated Cue docs) then this looks good to me. 👍🏻

Ah thanks, I forgot about the config schema output. I'll re-generate that now and verify that nothing changes.

**Comment by dsmith3197** [comment]
> > Assuming it passes CI and none of the configuration schema output changes (i.e. it doesn't change the generated Cue docs) then this looks good to me. 👍🏻
> 
> Ah thanks, I forgot about the config schema output. I'll re-generate that now and verify that nothing changes.

@tobz There were no config schema changes. I just had to update one test.

**Review by tobz — APPROVED** [review]

**Pull request merged**